### PR TITLE
gitserver: allow specifying build strategy

### PR DESCRIPTION
**Pull request by csrwng**
Adds environment variable to git server yaml to allow you to specify the strategy to use when invoking new-app. 

Fixes BZ 1336318

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal

File: examples/gitserver/README.md [hunk 1]
```diff
@@ -221,3 +221,24 @@ metadata:
...
+--------------
+
+When automatically starting a build, the git server will create a Docker type build if a Dockerfile is present
+in the repository. Otherwise, it will attempt a source type build. To force the git server to always use one
+strategy, set the BUILD_STRATEGY environment variable.
+
+Setting the BUILD_STRATEGY to `docker` will force new builds to be created with the Docker strategy:
+
+```sh
+oc set dc/git BUILD_STRATEGY=docker
```
Comment (bparees) [inline examples/gitserver/README.md]: oc set env ?

File: examples/gitserver/README.md [hunk 2]
```diff
@@ -221,3 +221,24 @@ metadata:
...
+```
+
+For OpenShift online which does not allow Docker type builds, you need to set the default strategy to `source`:
+
+```sh
+oc set dc/git BUILD_STRATEGY=source
```
Comment (bparees) [inline examples/gitserver/README.md]: oc set env i think

File: examples/gitserver/README.md (hunk 1 above)
Comment (csrwng) [inline examples/gitserver/README.md]: yes, that's what I meant, will fix

File: examples/gitserver/README.md [hunk 3]
```diff
@@ -221,3 +221,24 @@ metadata:
...
+in the repository. Otherwise, it will attempt a source type build. To force the git server to always use one
+strategy, set the BUILD_STRATEGY environment variable.
+
+Setting the BUILD_STRATEGY to `docker` will force new builds to be created with the Docker strategy:
+
+```sh
+oc set dc/git BUILD_STRATEGY=docker
+```
+
+For OpenShift online which does not allow Docker type builds, you need to set the default strategy to `source`:
```
Comment (bparees) [inline examples/gitserver/README.md]: it's not really a default, it's an override.
"For OpenShift Online which does not allow Docker type builds, you will need to set the strategy to `source` if your repository contains a Dockerfile:"

**Comment by csrwng** [comment]
updated

**Comment by bparees** [comment]
lgtm

**Comment by csrwng** [comment]
[merge]

**Comment by csrwng** [comment]
re-[merge]

**Comment by csrwng** [comment]
[merge]

**Pull request merged**